TAS_TBDAdapter/TASTBDAdapter.cs

```diff
@@ -43,15 +44,15 @@ public partial class TasTBDAdapter : BHoMAdapter
         [Input("tBDFilePath", "Path to tBD file")]
         [Input("tasSettings", "Input additional settings the adapter should use")]
         [Output("adapter", "Adapter to TAS tBD")]
-        public TasTBDAdapter(string tBDFilePath = "", TASSettings tasSettings = null)
+        public TasTBDAdapter(FileSettings fileSettings, TASSettings tasSettings = null)
         {
-            if(tasSettings == null)
+            if (tasSettings == null)
             {
                 BH.Engine.Reflection.Compute.RecordError("Please set some TAS Settings on the TAS Adapter");
                 return;
             }
-            //TBD application
-            tbdFilePath = tBDFilePath;
+            tbdFilePath = fileSettings.FileName;
```

Comment:
This should be `tbdFilePath = filesettings.GetFullFileName();`

Probably need to add `using BH.Engine.Adapter;` to the top as well.